Source/inv.h

```diff
@@ -78,54 +78,81 @@ extern bool invflag;
 extern bool drawsbarflag;
 extern const InvXY InvRect[73];
 
-void FreeInvGFX();
-void InitInv();
+class Inventory {
+
+public:
+	void FreeGFX();

```

Comment:
All the functions should be static since they don't change the class' instance.

Reply:
I thought about that but I think it would make derived classes more complex.

Perhaps, should I use this approach?

https://en.m.wikipedia.org/wiki/Curiously_recurring_template_pattern

The whole idea here is to make the UI more flexible (I plan other PRs), to allow couch coop.

```
Inventory inventory = isCouchCoop ? new CouhCoopInventory() : new Inventory();
```

I am open to suggestions, I just wanted to make the changes simple because couch coop requires lots of changes.

Reply:
So the idea here is to allow to override certain inventory functions? In that case don't make the functions static, no. I'm not a huge fan of inheritance in general, but it all depends on what you want to do so it's difficult to advise now.